epochStart/bootstrap/process.go

```diff
@@ -593,6 +597,9 @@ func (e *epochStartBootstrap) processNodesConfig(pubKey []byte) error {
 	}
 
 	e.nodesConfig, e.baseData.shardId, err = e.nodesConfigHandler.NodesConfigFromMetaBlock(e.epochStartMeta, e.prevEpochStartMeta)
+	if e.destinationShardAsObserver != core.DefaultShardIDAsObserver {
+		e.baseData.shardId = e.destinationShardAsObserver
```

Comment:
this does not sound good - if the node is validator it shouldn't rewrite it to desired observer state.